config-common/src/watcher.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use async_trait::async_trait;
+use notify::Watcher;
+use std::path::PathBuf;
+
+type Api<TConfig, TError> = std::sync::Arc<
+    futures_util::lock::Mutex<dyn UpdateConfig<Config = TConfig, Error = TError> + Send>,
+>;
+
+#[derive(Debug)]
+pub enum ReprovisionTrigger {
```

Comment:
1. Only IS considers these to be "reprovision triggers". The other services have no concept of provisioning.

1. `Api` and `Startup` are only used by IS. `Api` in particular only makes sense for IS.

1. Most importantly, the file watcher is the only caller of `update_config` except for one case where it's called in IS startup.

I suggest keeping this type IS-specific and removing it from `update_config`. For IS, you can have another function that does the config update and have it take a `ReprovisionTrigger`, and call it from IS's `update_config` impl with `ReprovisionTrigger::ConfigFileUpdate`.

Reply:
Fixed